packages/runtime-core/src/hydration.ts

```diff
@@ -129,7 +135,7 @@ export function createHydrationFunctions(
             nextNode = onMismatch()
           }
         } else {
-          if ((node as Text).data !== vnode.children) {
+          if ((node as Text).data !== escapeHtml(vnode.children)) {
```

Comment:
I think this will break normal text node hydration for anything that contains escaped characters. e.g.:

```html
<template>
  <hr>
  Hello &amp; World
  <hr>
</template>
```

[SFC Playground using this PR](https://deploy-preview-7776--vue-sfc-playground.netlify.app/#__SSR__eNqljTuOwjAQhq9iTbEVY3cbKRuttN3egMZNIAME+TEaO0ERyt1xEAIEJdVo/td3hj9mPQ4ENTSZPLs2068NSjUHud5/ci6qr9bzj1pHcd3DbMy9ASvoPUfJ6FvWxxRDGTwvUXszkoVaXZVFK8Tlt3DImVNtTEfs4oQsNPZ0wqqqvhFLCtNui4Ux7SUOodOBsut3k26ZTbG1DCH3njQljxuJp0RS8BZWTyRTxJEEhUJHQvI5+WXwjb7AZxtmmC/L53qt)